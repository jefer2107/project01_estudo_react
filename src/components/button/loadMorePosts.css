.button{
    width: 100%;
    padding: 15px 30px;
    background-color: blue;
    color:#fff;
    font-weight: bold;
}

.button:disabled{
    background-color: #ddd;
    cursor: not-allowed;
}