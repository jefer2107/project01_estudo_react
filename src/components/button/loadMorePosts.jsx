import './loadMorePosts.css';
import p from 'prop-types';

export const LoadMorePosts = (props) => {
  const { loadMorePosts, text, disabled } = props;

  return (
    <button disabled={disabled} className="button" onClick={loadMorePosts}>
      {text}
    </button>
  );
};

LoadMorePosts.defaultProps = {
  disabled: false,
};

LoadMorePosts.prototype = {
  text: p.string,
  loadMorePosts: p.func.isRequired,
  disabled: p.bool,
};
