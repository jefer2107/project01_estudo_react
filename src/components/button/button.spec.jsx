import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { LoadMorePosts } from './loadMorePosts';

describe('/<LoadMorePosts />', () => {
  it('should render the button with the text', () => {
    render(<LoadMorePosts text="Load More Posts" />);
    expect.assertions(1);

    const button = screen.getByRole('button', { name: /load more posts/i });
    expect(button).toBeInTheDocument();
  });

  it('should call function on button click', () => {
    const fn = jest.fn();

    render(<LoadMorePosts text="Load More Posts" onClick={fn} />);

    const button = screen.getByRole('button', { name: /load more posts/i });

    userEvent.click(button);

    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('should be disabled when disabled is true', () => {
    render(<LoadMorePosts text="Load More Posts" disabled={true} />);

    const button = screen.getByRole('button', { name: /load more posts/i });

    expect(button).toBeDisabled();
  });

  it('should be enabled when disabled is false', () => {
    render(<LoadMorePosts text="Load More Posts" disabled={false} />);

    const button = screen.getByRole('button', { name: /load more posts/i });

    expect(button).toBeEnabled();
  });

  it('should match snapshots', () => {
    const fn = jest.fn();
    const { container } = render(<LoadMorePosts text="Load More Posts" disabled={false} onClick={fn} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
