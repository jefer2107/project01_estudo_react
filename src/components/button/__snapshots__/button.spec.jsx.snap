// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/<LoadMorePosts /> should match snapshots 1`] = `
<button
  class="button"
>
  Load More Posts
</button>
`;
